apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts

```diff
@@ -149,15 +148,22 @@ export class ExpandEmailEditorSchemaUsecase {
    * ]
    *
    */
-  private multiplyForEachNode(
+  private async multiplyForEachNode(
     node: TipTapNode & { attrs: { [MailyAttrsEnum.EACH_KEY]: string } },
     variables: FullPayloadForRender
-  ): TipTapNode[] {
+  ): Promise<TipTapNode[]> {
     const iterablePath = node.attrs[MailyAttrsEnum.EACH_KEY];
     const nodeContent = node.content || [];
     const expandedContent: TipTapNode[] = [];
 
-    const iterableArray = this.getValueByPath(variables, iterablePath);
+    const iterableArrayString = await parseLiquid(iterablePath, variables);
```

Comment:
Use Liquid engine to extract the array from payload instead of finding it manually.